src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs

```diff
@@ -450,13 +466,47 @@ private static async Task<Solution> MergeDocumentChangesAsync(Solution originalS
                 cancellationToken: cancellationToken).ConfigureAwait(false);
 
             var referencedSymbols = progress.GetReferencedSymbols();
-            return referencedSymbols.Where(refSymbol => refSymbol.Definition.Equals(symbol))
-                    .SelectMany(refSymbol => refSymbol.Locations).ToImmutableArray();
+            builder.AddRange(referencedSymbols
+                .Where(refSymbol => refSymbol.Definition.Equals(symbol))
+                .SelectMany(refSymbol => refSymbol.Locations)
+                .Select(location => new LocationForAffectedSymbol(location, false)));
+
+            // Find references above doesn't handle invocation of extension methods (in reduced form)
+            // that are declared in the affected types, so we need to find references to extension methods
+            // as well. This will returns all the references, not just in the reduced form. But we will
+            // not further distinguish the usage. In the worst case, those references are redundant because
+            // they could be covered by the type references found above.
+            if (symbol is INamedTypeSymbol typeSymbol && typeSymbol.MightContainExtensionMethods)
+            {
+                foreach (var memberSymbol in typeSymbol.GetMembers())
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    if (memberSymbol is IMethodSymbol methodSymbol && methodSymbol.IsExtensionMethod)
+                    {
+                        var methodProgress = new StreamingProgressCollector(StreamingFindReferencesProgress.Instance);
+                        await SymbolFinder.FindReferencesAsync(
+                            symbolAndProjectId: SymbolAndProjectId.Create(methodSymbol, document.Project.Id),
+                            solution: document.Project.Solution,
+                            documents: null,
+                            progress: methodProgress,
+                            options: FindReferencesSearchOptions.Default,
+                            cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
nit: extract out this code so you can call it above and here in a consistent fashion.